src/classes/ADDR_CannotDelete_TEST.cls

```diff
@@ -0,0 +1,452 @@
+        }
+    }
+
+    /*********************************************************************************************************
+    * @description Test method for the hasChildRecords() method in ADDR_CannotDelete_TDTM to test that TRUE is
+    * returned when an Address record has child records of only type Account.
+    */
+    @isTest
+    private static void testAddrHasChildAccountRecordsIsTrue() {
+
+        UTIL_CustomSettingsFacade.getSettingsForTests(new Hierarchy_Settings__c
+                                                            (Account_Processor__c = ADDR_CannotDelete_TEST.hhAccRecordTypeID));
+
+        UTIL_UnitTestData_TEST.AccsWithAddrs addrsWithRelatedAcc = UTIL_UnitTestData_TEST.createTestAccsAddrs(5, ADDR_CannotDelete_TEST.hhAccRecordTypeID);
+        
+        List<Address__c> addrsWithChildAcc = addrsWithRelatedAcc.addrs; 
+
+        List<Address__c> returnAddresses = [
+            SELECT Id, 
+                (SELECT Id FROM Address__c.Accounts__r)
+            FROM Address__c
+            WHERE Id IN :addrsWithChildAcc
+        ];
+
+        ADDR_CannotDelete_TDTM addrCannotDeleteClass = new ADDR_CannotDelete_TDTM();
+
+        for (Address__c addr : returnAddresses) {
+            System.assert(addrCannotDeleteClass.hasChildRecords(addr) == TRUE);
+        }
+    }
+
+    /*********************************************************************************************************
+    * @description Test method for the hasChildRecords() method in ADDR_CannotDelete_TDTM to test that TRUE is
+    * returned when an Address record has both Contact and Account child records.
+    */
+    @isTest
+    private static void testAddrHasChildRecordsIsTrue() {
+
+        UTIL_CustomSettingsFacade.getSettingsForTests(new Hierarchy_Settings__c
+                                                            (Account_Processor__c = ADDR_CannotDelete_TEST.hhAccRecordTypeID));
+
+        UTIL_UnitTestData_TEST.AccsWithAddrs addrsWithRelatedAcc = UTIL_UnitTestData_TEST.createTestAccsAddrs(5, ADDR_CannotDelete_TEST.hhAccRecordTypeID);
+        UTIL_UnitTestData_TEST.ContactsWithAddrs addrsWithRelatedContact = UTIL_UnitTestData_TEST.createTestContactsAddrs(5);
+
+        List<Address__c> addrsWithChildRecords = new List<Address__c>();
+        addrsWithChildRecords.addAll(addrsWithRelatedAcc.addrs); 
+        addrsWithChildRecords.addAll(addrsWithRelatedContact.addrs);
+
+        List<Address__c> returnAddresses = [
+            SELECT Id, 
+                (SELECT Id FROM Address__c.Contacts1__r),
+                (SELECT Id FROM Address__c.Accounts__r)
+            FROM Address__c
+            WHERE Id IN :addrsWithChildRecords
+        ];
+
+        ADDR_CannotDelete_TDTM addrCannotDeleteClass = new ADDR_CannotDelete_TDTM();
+        
+        for (Address__c addr : returnAddresses) {
+            System.assert(addrCannotDeleteClass.hasChildRecords(addr) == TRUE);
```

Comment:
@m-kinkade No changes here. This communicated your intent clearly, and I wanted to call that out. If you believe that adding == TRUE or == FALSE may effectively communicate your intent in the assertion, you may want to consider the readability of System.assertEquals.

Reply:
Great suggestion.  I've updated my assertions to use System.assertEquals because I agree that it is more readable in most cases.